src/libstd/sys/windows/fs.rs

```diff
@@ -377,8 +377,11 @@ impl fmt::Debug for File {
     }
 }
 
-pub fn to_utf16(s: &Path) -> Vec<u16> {
-    s.as_os_str().encode_wide().chain(Some(0)).collect()
+pub fn to_u16s(s: &Path) -> io::Result<Vec<u16>> {
+    if s.as_os_str().to_inner().to_inner().iter().any(|&b| b == 0) {
```

Comment:
A feature of the old and new UTF-8 (the old UTF-8 corresponds to our WTF-8) is that all ASCII codepoint and only the ASCII codepoints are represented by the bytes 0 to 127. I can still change that if you want, though.